python/lsst/dm/OCPS/ocps_csc.py

```diff
@@ -0,0 +1,230 @@
+
+    async def do_execute(self, data):
+        """Implement the ``execute`` command.
+
+        Submit a request for execution to the back-end REST API.
+
+        Parameters
+        ----------
+        data: types.SimpleNamespace
+            Must contain pipeline, version, and config attributes
+        """
+        self.assert_enabled("execute")
+        self.log.info(f"execute command with {data}")
+
+        if self.simulation_mode == 0:
+            payload = dict(
+                type="pipeline",
+                version=data.version,
+                config=data.config,
+                env=[dict(name="PIPELINE", value=data.pipeline)],
+            )
+            self.log.info(f"PUT: {payload}")
+            result = self.connection.put(self.config.url, json=payload)
+            result.raise_for_status()
+            self.log.info(f"PUT result: {result.data}")
+            job_id = result.json().job_id
+            status_url = f"{self.config.url}/{job_id}"
+        else:
+            if data.pipeline not in ("true.yaml", "false.yaml", "fault.yaml"):
+                raise salobj.ExpectedError(f"Unknown pipeline: {data.pipeline}")
+            job_id = f"{data.pipeline}-{time.time()}"
+            status_url = f"ocps://{job_id}"
+            self.log.info(f"Simulated POST result: {status_url}")
+            self.simulated_jobs.add(job_id)
+
+        payload = json.dumps(dict(job_id=job_id))
+        self.cmd_execute.ack_in_progress(data, timeout=600.0, result=payload)
+        self.log.info(f"Ack in progress: {payload}")
+        self.log.info(f"Starting async wait: {status_url}")
+
+        while True:
+            if status_url.startswith("ocps://"):
+                await asyncio.sleep(abs(random.normalvariate(10, 4)))
+                # simulation mode
+                self.log.info(f"Simulating result for {job_id}")
+                if job_id not in self.simulated_jobs:
+                    raise salobj.ExpectedError(f"No such job id: {job_id}")
+                self.simulated_jobs.remove(job_id)
+                if job_id.startswith("true.yaml-"):
+                    payload = json.dumps(dict(result=True))
+                    self.evt_job_result.set_put(
+                        job_id=job_id, exit_code=0, result=payload
+                    )
+                elif job_id.startswith("false.yaml-"):
+                    payload = json.dumps(dict(result=False))
+                    self.evt_job_result.set_put(
+                        job_id=job_id, exit_code=0, result=payload
+                    )
+                elif job_id.startswith("fault.yaml-"):
+                    self.fault(
+                        code=2, report="404 Simulation cannot contact execution service"
+                    )
+                    raise salobj.ExpectedError("Failed to connect (simulated)")
+                else:
+                    raise salobj.ExpectedError(f"Unknown pipeline: {job_id}")
```

Comment:
Please tweak this to indicate that the CSC was running the simulation branch (which is not simulation_mode = 1 so I don't know what to call it!). Unknown simulation pipeline?